### PR TITLE
Add config option for zip extract charset

### PR DESCRIPTION
**Pull request by SammyIsConfused**
##### Checklist

- [x] the [contributor license agreement][] is signed
- [x] commit message follows [commit guidelines][]
- [x] documentation has been added [(This PR in openequella.github.io)](https://github.com/openequella/openequella.github.io/pull/70)
##### Description of change
#1868 
With this change, adding `filestore.zipExtractCharset` in the `optional-config.properties` will set zip extraction to use that charset. 
For example, for issue #1868 setting this to be `filestore.zipExtractCharset = ISO_8859_1` makes the special character SCORM package work as normal. 

[contributor license agreement]: https://www.apereo.org/node/676
[commit guidelines]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: Platform/Plugins/com.tle.platform.common/src/com/tle/core/util/archive/ArchiveType.java [hunk 1]
```diff
@@ -124,12 +123,13 @@ public ArchiveCreator createArchiver(OutputStream archive) throws IOException {
...
   private final String[] fileExtensions;
 
-  private ArchiveType(String... fileExtensions) {
+  ArchiveType(String... fileExtensions) {
     assert fileExtensions != null;
     this.fileExtensions = fileExtensions;
   }
 
-  public abstract ArchiveExtractor createExtractor(InputStream archive) throws IOException;
+  public abstract ArchiveExtractor createExtractor(InputStream archive, String charset)
```
Comment (edalex-ian) [inline Platform/Plugins/com.tle.platform.common/src/com/tle/core/util/archive/ArchiveType.java:131]: Down at this level we should be using the tighter type of `Charset`.

File: Source/Plugins/Core/com.equella.core/src/com/tle/core/filesystem/guice/FilesystemModule.java [hunk 2]
```diff
@@ -34,6 +34,7 @@
   protected void configure() {
     bindBoolean("files.useXSendfile");
     bindBoolean("filestore.advanced");
+    bindProp("filestore.zipExtractCharset", "UTF-8");
```
Comment (edalex-ian) [inline Source/Plugins/Core/com.equella.core/src/com/tle/core/filesystem/guice/FilesystemModule.java:37]: Similar to what you've done elsewhere, it'd be better here to use `StandardCharsets.UTF_8.name()`.

**Comment by edalex-ian** [comment]
I'm feeling the build issues you're seeing here are related to https://github.com/actions/checkout/issues/299

**Comment by SammyIsConfused** [comment]
Okay, build issues fixed thanks to @edalex-ian reverting us to `@checkout/v1`. Ready to go after a final once-over.

**Pull request merged**